packages/docprovider/src/ydrive.ts

```diff
@@ -220,6 +270,29 @@ export class YDrive extends Drive implements ICollaborativeDrive {
         }
         this._globalAwareness?.setLocalStateField('documents', documents);
       });
+
+      const notebookPath = URLExt.join(
+        this.serverSettings.baseUrl,
+        DOCUMENT_TIMELINE_URL,
+        options.path
+      );
+      if (!this._timelineWidget) {
+        this._timelineWidget = new TimelineWidget(
+          notebookPath,
+          provider,
+          options.contentType,
+          options.format
+        );
+      } else {
+        this.updateTimelineForNotebook(options.path);
+        this._timelineWidget.update();
+      }
+      const elt = document.getElementById('slider-status-bar');
+      if (elt && !this._timelineWidget.isAttached) {
+        Widget.attach(this._timelineWidget, elt);
```

Comment:
Conceptually, it is not a responsibility of the drive to know anything about widgets. Usually we would create a new "model" (like `timelineModel`) to interface between one object knowing about multiple things and the widgets, but here a simpler solution might be possible.

Practically, querying by ID is a code smell.

Here is a proposal:
- let's move instationation and attachement of `TimelineWidget` to the `@jupyter/docprovider-extension:statusBarTimeline` plugin (where it will be a breeze because you have the status bar element there widget reference)
- replace `updateTimelineForNotebook` method on `YDrive` with `getProviderForPath(path: string): Promise<IForkProvider>` where:
  ```ts
  interface IForkProvider {
    connectToFork: (arguments) => Promise<someType>;
    contentType: string;
    format: string;
  }
  ```
  and mark `WebSocketProvider` as implementing `IForkProvider` interface
- if absolutely needed add a signal on `YDrive` such as `modelCreated: Signal<YDrive, IModelCreatedArgs>` so that you can call `getProviderForPath()` immediately once this `_onCreate()` is called, where:
  ```ts
  interface IModelCreatedArgs {
     options: Contents.ISharedFactoryOptions;
     sharedModel: YDocument<DocumentChange>;
  }
  ```
   but maybe you could just use document tracker instead.

Reply:
Maybe `getProviderForPath` → `getForkProvider`?